src/Endpoints/AbstractEndpoint.php

```diff
@@ -42,18 +41,20 @@ public function getCollectionName(): string
 
     protected function queryBuild(array $params): string
     {
-        if ($params) {
-            return '?' . Query::build($params);
+        $p = [];
+        foreach ($params as $k => $v) {
+            $p[] = urldecode($k) . "=" . urlencode($v);
         }
-        return '';
+
+        return '?' . implode('&', $p);
     }
 
     /**
      * @throws InvalidArgumentException
      */
     protected function createRequest(string $method, string $uri, array $body = []): RequestInterface
     {
-        $httpFactory = new HttpFactory();
+        $httpFactory = Psr17FactoryDiscovery::findRequestFactory();
```

Comment:
What if I want to use my own http factory implementation?

Reply:
Changed this part with a getter and setter as below: 

```php
    public function getHttpFactory(): RequestFactoryInterface
    {
        if ($this->httpFactory === null) {
            $this->httpFactory = Psr17FactoryDiscovery::findRequestFactory();
        }
        return $this->httpFactory;
    }

    public function setHttpFactory(RequestFactoryInterface $httpFactory): void
    {
        $this->httpFactory = $httpFactory;
    }
```